### PR TITLE
Initial DocSearch demo notebook

### PR DESCRIPTION
**Pull request by JoshEngels**

### CODE REVIEW

File: DocSearch.ipynb [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is a demonstration for how to use ThirdAI's document search engine. It assumes you are running in the ThirdAI demo docker container. Once in the container, you can get this notebook by running\n",
+    "```\n",
+    "wget https://raw.githubusercontent.com/ThirdAILabs/Demos/main/DocSearch.ipynb\n",
```
Comment (benitoThree) [inline DocSearch.ipynb:9]: is this not going to run automatically? Do you think we can just have a ./download_notebook.sh?

File: DocSearch.ipynb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   "source": [
+    "Now let's create an initially empty index! As a quick description of the input to the constructor, \n",
+    "* dense_input_dimension is the dimension of the output of our embedding model (128 with our preloaded model)\n",
+    "\n",
+    "* num_tables and hashes_per_table are hyperparameters of the index\n",
+    "  * increasing num_tables increases the accuracy of the model at the cost of speed and memory (a good value is the prepopulated 16)\n",
+    "  \n",
+    "  * hashes_per_table has a sweet spot in accuracy at around log_2(average document size)\n",
+    "  \n",
+    "* centroids are a numpy array that represents precomputed centroids for the embedding space; we've already calculated and stored these for you! We've precomputed 2^18 of them, but if you're only adding a few points a simple heuristic is to select a random subsample of them, which we do below."
```
Comment (benitoThree) [inline DocSearch.ipynb:68]: I thought we have an api that doesn't take in centroids?

File: DocSearch.ipynb [hunk 3]
```diff
@@ -0,0 +1,145 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is a demonstration for how to use ThirdAI's document search engine. It assumes you are running in the ThirdAI demo docker container. Once in the container, you can get this notebook by running\n",
```
Comment (benitoThree) [inline DocSearch.ipynb:7]: Might be nice to list the sections of this notebook, e.g. 1. generating centroids, 2. building index, etc. just to give a high level overview of what's involved in building this system

File: DocSearch.ipynb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    "from thirdai.search import DocRetrieval\n",
+    "\n",
+    "from embeddings import DocSearchModel\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " Now you're going to need a dataset, which just consists of a collection of (document_id, document_text) pairs. Our engine works best with document sizes between 20 and 200 words long. Feel free to split big documents up into many small passages!"
```
Comment (benitoThree) [inline DocSearch.ipynb:43]: Nit: Maybe not start every markdown with "now"?

File: DocSearch.ipynb (hunk 3 above)
Comment (JoshEngels) [reply DocSearch.ipynb:7]: Sure, good idea

File: DocSearch.ipynb (hunk 2 above)
Comment (JoshEngels) [reply DocSearch.ipynb:68]: Nope, not yet (but we pregenerate centroids for people)

File: DocSearch.ipynb (hunk 1 above)
Comment (JoshEngels) [reply DocSearch.ipynb:9]: I think this is fine for now, I want to get something out without another pr on the container side

**Pull request merged**